### PR TITLE
[#4115]  feat(api):  add validate check for decimal(0,0)

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
add validate check for decimal(0,0)

### Why are the changes needed?

Fix: #4115 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
existing tests

### CODE REVIEW

File: api/src/main/java/org/apache/gravitino/rel/types/Types.java
Comment (mchades) [inline api/src/main/java/org/apache/gravitino/rel/types/Types.java:1]: Could you also add the limitations to the doc `manage-relational-metadata-using-gravitino.md#gravitino-table-column-type`?

Comment (FANNG1) [reply api/src/main/java/org/apache/gravitino/rel/types/Types.java:1]: ok

Comment (FANNG1) [reply api/src/main/java/org/apache/gravitino/rel/types/Types.java:1]: updated

**Review by mchades — APPROVED** [review]
LGTM.

**Pull request merged**